[openbsd] BaseLogFile.cc:244:61: error: invalid conversion from 'long int*' to 'time_t* {aka long long int*}' [-fpermissive]

Build fails on BaseLogFile.cc.

```
$ sysctl kern.version
kern.version=OpenBSD 6.0-current (GENERIC.MP) #117: Sat Jan  7 09:10:45 MST 2017
    build@amd64.openbsd.org:/usr/src/sys/arch/amd64/compile/GENERIC.MP
```

```
Making all in ts
gmake[3]: Entering directory '/home/jirib/openbsd/pobj/trafficserver-20170106/trafficserver-24347df553e91660b794af1079fbf69f986dc5b4/lib/ts'
depbase=`echo BaseLogFile.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/usr/bin/libtool  --tag=CXX   --mode=compile c++ -DHAVE_CONFIG_H -I. -I../../lib  -I/home/jirib/openbsd/pobj/trafficserver-20170106/trafficserver-24347df553e91660b794af1079fbf69f986dc5b4/lib -I/usr/local/include
-Dopenbsd -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_GNU_SOURCE -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -D_FORTIFY_SOURCE=2 -I/usr/include -DOPENSSL_NO_SSL_INTERN -I/usr/local/include/tcl8.5  -O2 -pipe -O0 -g -std=c++11 -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -fPIE -fstack-protector -mcx16 -MT BaseLogFile.lo -MD -MP -MF $depbase.Tpo -c -o BaseLogFile.lo BaseLogFile.cc &&\
mv -f $depbase.Tpo $depbase.Plo
c++ -DHAVE_CONFIG_H -I. -I../../lib -I/home/jirib/openbsd/pobj/trafficserver-20170106/trafficserver-24347df553e91660b794af1079fbf69f986dc5b4/lib -I/usr/local/include -Dopenbsd -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_GNU_SOURCE -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -D_FORTIFY_SOURCE=2 -I/usr/include -DOPENSSL_NO_SSL_INTERN -I/usr/local/include/tcl8.5 -O2 -pipe -O0 -g -std=c++11 -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -fstack-protector -mcx16 -MT BaseLogFile.lo -MD -MP -MF .deps/BaseLogFile.Tpo -c BaseLogFile.cc -fPIC -DPIC -o .libs/BaseLogFile.o
BaseLogFile.cc: In member function 'int BaseLogFile::roll()':
BaseLogFile.cc:244:61: error: invalid conversion from 'long int*' to 'time_t* {aka long long int*}' [-fpermissive]
   if (!m_meta_info || !m_meta_info->get_creation_time(&start))
                                                             ^
In file included from BaseLogFile.cc:24:0:
BaseLogFile.h:126:3: note: initializing argument 1 of 'bool BaseMetaInfo::get_creation_time(time_t*)'
   get_creation_time(time_t *time)
   ^
Error while executing c++ -DHAVE_CONFIG_H -I. -I../../lib -I/home/jirib/openbsd/pobj/trafficserver-20170106/trafficserver-24347df553e91660b794af1079fbf69f986dc5b4/lib -I/usr/local/include -Dopenbsd -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_GNU_SOURCE -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -D_FORTIFY_SOURCE=2 -I/usr/include -DOPENSSL_NO_SSL_INTERN -I/usr/local/include/tcl8.5 -O2 -pipe -O0
-g -std=c++11 -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -fstack-protector -mcx16 -MT BaseLogFile.lo -MD -MP -MF .deps/BaseLogFile.Tpo -c BaseLogFile.cc -fPIC -DPIC -o .libs/BaseLogFile.o
```

--- a/lib/ts/BaseLogFile.cc
+++ b/lib/ts/BaseLogFile.cc
@@ -238,7 +238,7 @@ BaseLogFile::roll(long interval_start, long interval_end)
 int
 BaseLogFile::roll()
 {
-  long start;
+  time_t start;
   time_t now = time(nullptr);
 
   if (!m_meta_info || !m_meta_info->get_creation_time(&start))

